src/arrayMethodReduce.js

```diff
@@ -5,7 +5,21 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let acc;
+    let loopStart = 0;
+
+    if (initialValue !== undefined) {
+      acc = initialValue;
+    } else {
+      acc = this[0];
+      loopStart++;
+    }
```

Comment:
```suggestion
    let acc = initialValue;
    let loopStart = 0;

    if (initialValue === undefined) {
      acc = this[0];
      loopStart = 1;
    }
```

The logic is correct, but let's make your code clearer and get rid of unnecessary 'else'.
You need to change your 'acc' and 'loopStart' variables only if 'intialValue' is undefined. Only one check.